x/programs/rust/examples/single_nft/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+        .store(StateKey::TotalSupply.to_vec(), &TOTAL_SUPPLY)
+        .expect("failed to store total supply");
+
+    // Initialize counter
+    program
+        .state()
+        .store(StateKey::Counter.to_vec(), &0)
+        .expect("failed to store counter");
+
+    true
+}
+
+/// Mints NFT tokens and sends them to the recipient.
+#[public]
+pub fn mint(program: Program, recipient: Address) -> bool {
+    const MINT_AMOUNT: i64 = 1;
+
+    let counter = program
+        .state()
+        .get::<i64, _>(StateKey::Counter.to_vec())
+        .expect("failed to store ");
+
+    assert!(
+        counter <= TOTAL_SUPPLY as i64,
+        "max supply for nft exceeded"
+    );
+
+    let balance = program
+        .state()
+        .get::<i64, _>(StateKey::Balance(recipient).to_vec())
+        .expect("failed to get balance");
+
+    program
+        .state()
+        .store(
+            StateKey::Balance(recipient).to_vec(),
+            &(balance + MINT_AMOUNT),
+        )
+        .expect("failed to store balance");
+
+    program
+        .state()
+        .store(StateKey::Owner.to_vec(), &recipient)
+        .expect("failed to store owner");
+
+    program
+        .state()
+        .store(StateKey::Counter.to_vec(), &(counter + 1))
+        .is_ok()
+}
+
+#[public]
+pub fn burn(program: Program, from: Address) -> bool {
+    const BURN_AMOUNT: i64 = 1;
+
+    // Only the owner of the NFT can burn it
+    let owner = program
+        .state()
+        .get::<Address, _>(StateKey::Owner.to_vec())
+        .expect("failed to get owner");
+
+    assert_eq!(owner, from, "only the owner can burn the nft");
+
+    let balance = program
+        .state()
+        .get::<i64, _>(StateKey::Balance(from).to_vec())
+        .expect("failed to get balance");
+
+    assert!(
+        BURN_AMOUNT <= balance,
+        "amount burned must be less than or equal to the user balance"
+    );
+
+    let counter = program
+        .state()
+        .get::<i64, _>(StateKey::Counter.to_vec())
+        .expect("failed to get counter");
+
+    assert!(counter > 0, "cannot burn more nfts");
+
+    // Burn the NFT by transferring it to the zero address
+    program
+        .state()
+        .store(StateKey::Balance(from).to_vec(), &(balance - BURN_AMOUNT))
+        .expect("failed to store new balance");
+
+    // TODO move to a lazy static? Or move to the VM layer entirely
+    let null_address = Address::new([0; 32]);
+    program
+        .state()
+        .store(StateKey::Owner.to_vec(), &null_address)
+        .is_ok()
+}
+
+#[cfg(test)]
+mod tests {
+    use serial_test::serial;
+    use std::env;
+    use wasmlanche_sdk::simulator::{
+        self, id_from_step, Key, Operator, PlanResponse, Require, ResultAssertion,
+    };
+
+    use crate::example;
+
+    // export SIMULATOR_PATH=/path/to/simulator
+    // export PROGRAM_PATH=/path/to/program.wasm
+    // cargo cargo test --package token --lib nocapture -- tests::test_token_plan --exact --nocapture --ignored
```

Comment:
It would be better if you check these variables at compile time with the `std::env` macro instead of forcing the user to use `--ignored`

Reply:
Also cargo echo
```suggestion
    // cargo test --package token --lib nocapture -- tests::test_token_plan --exact --nocapture --ignored
```

Reply:
Is this command correct?

Reply:
I really don't like that we comment how to run tests in the test file itself for this very reason. It's giving a person a fish instead of teaching them how to fish...

Reply:
comment here is wrong

Reply:
I think this comment is pointless and will remove it entirely. It's already in main for the token example however.